### PR TITLE
Package dkml-component-staging-ocamlrun.4.12.1~v1.0.0

### PR DESCRIPTION
**Pull request by jonahbeckford**
### `dkml-component-staging-ocamlrun.4.12.1~v1.0.0`
DKML component for ocamlrun
Standalone distribution of OCaml in <share>/staging-files containing just ocamlrun, the OCaml Stdlib and the other OCaml libraries (str, unix, bigarray, etc.)

---
* Homepage: https://github.com/diskuv/dkml-component-ocamlrun
* Source repo: git+https://github.com/diskuv/dkml-component-ocamlrun.git
* Bug tracker: https://github.com/diskuv/dkml-component-ocamlrun/issues

---
:camel: Pull-request generated by opam-publish v2.1.0

### CODE REVIEW

**Comment by jonahbeckford** [comment]
I'll be adding some `available:` constraints to fix the CI.

**Comment by jonahbeckford** [comment]
This is ready to merge. Thanks!!

**Comment by kit-ty-kate** [comment]
There seems to be an issue on macOS:
```
# patches(ocaml cross) = /Users/mac1000/.opam/default/.opam-switch/build/dkml-component-staging-ocamlrun.4.12.1~v1.0.0/dkmldir/vendor/dkml-compiler/src/p/ocaml-common-4_12-a01-alignfiletime.patch /Users/mac1000/.opam/default/.opam-switch/build/dkml-component-staging-ocamlrun.4.12.1~v1.0.0/dkmldir/vendor/dkml-compiler/src/p/ocaml-cross-4_12-a01.patch /Users/mac1000/.opam/default/.opam-switch/build/dkml-component-staging-ocamlrun.4.12.1~v1.0.0/dkmldir/vendor/dkml-compiler/src/p/ocaml-cross-4_12-a02-arm32.patch
# Applying: Diskuv ocaml cross patch ocaml-common-4_12-a01-alignfiletime.patch
# Applying: Diskuv ocaml cross patch ocaml-cross-4_12-a01.patch
# Applying: Diskuv ocaml cross patch ocaml-cross-4_12-a02-arm32.patch
# patches(flexdll cross) = /Users/mac1000/.opam/default/.opam-switch/build/dkml-component-staging-ocamlrun.4.12.1~v1.0.0/dkmldir/vendor/dkml-compiler/src/p/flexdll-cross-0_39-a01-arm64.patch
# Applying: Diskuv flexdll cross patch flexdll-cross-0_39-a01-arm64.patch
# ==> Tapping homebrew/bundle
# fatal: could not create work tree dir '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-bundle': Operation not permitted
# Error: Failure while executing; `git clone https://github.com/Homebrew/homebrew-bundle /usr/local/Homebrew/Library/Taps/homebrew/homebrew-bundle --origin=origin --template=` exited with 128.
# Error: Failure while executing; `/usr/local/bin/brew tap homebrew/bundle` exited with 1.
```

**Comment by kit-ty-kate** [comment]
You can fix the issue by exporting `HOMEBREW_NO_AUTO_UPDATE=1` in your scripts calling `brew`

File: packages/dkml-component-staging-ocamlrun/dkml-component-staging-ocamlrun.4.12.1~v1.0.0/opam [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+}
+extra-source "dl/dkml-compiler.tar.gz" {
+  src: "https://github.com/diskuv/dkml-compiler/archive/refs/tags/4.12.1-v1.0.0.tar.gz"
+  checksum: [
+    "sha256=b4850314ea64d357ce814faebc1f1ef54052e76ea048602401f481cf6bf40455"
+  ]
+}
+extra-source "dl/dkml-runtime-common.tar.gz" {
+  # This violates the v1.0.0 versioning, but this future patch is needed to resolve build
+  # issues on macOS in Opam CI.
```
Comment (kit-ty-kate) [inline packages/dkml-component-staging-ocamlrun/dkml-component-staging-ocamlrun.4.12.1~v1.0.0/opam:157]:
```suggestion
  # issues on macOS in Opam.
```

**Comment by jonahbeckford** [comment]
This is also done. Thanks!

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**